openmls/src/group/core_group/proposals.rs

```diff
@@ -166,6 +166,38 @@ impl QueuedProposal {
     }
 }
 
+/// Helper struct to collect proposals such that they are unique and can be read
+/// out in the order in that they were added.
+struct OrderedProposalRefs {
+    proposal_refs: HashSet<ProposalRef>,
+    ordered_proposal_refs: Vec<ProposalRef>,
+}
+
+impl OrderedProposalRefs {
+    fn new() -> Self {
+        Self {
+            proposal_refs: HashSet::new(),
+            ordered_proposal_refs: Vec::new(),
+        }
+    }
+
+    /// Adds a proposal reference to the queue. If the proposal reference is
+    /// already in the queue, it ignores it.
+    fn insert(&mut self, proposal_ref: ProposalRef) {
```

Comment:
`insert` is the wrong verb here, because it suggests new values can be inserted in the middle, like on a `HashMap`. Maybe something like `add_if_unique`?

Reply:
I just called it `add` now. The uniqueness part should be clear from the nature of the type.